Components/Sensors/BatteryTask.cpp

```diff
@@ -0,0 +1,193 @@
+    //Ensure creation succeded
+    SOAR_ASSERT(rtValue == pdPASS, "BatteryTask::InitTask() - xTaskCreate() failed");
+}
+
+/**
+ * @brief BatteryTask run loop
+ * @param pvParams Currently unused task context
+ */
+void BatteryTask::Run(void * pvParams)
+{
+    while (1) {
+        Command cm;
+
+        //Wait forever for a command
+        qEvtQueue->ReceiveWait(cm);
+
+        //Process the command
+        HandleCommand(cm);
+    }
+}
+
+/**
+ * @brief Handles a command
+ * @param cm Command reference to handle
+ */
+void BatteryTask::HandleCommand(Command& cm)
+{
+    //TODO: Since this task will stall for a few milliseconds, we may need a way to eat the whole queue (combine similar eg. REQUEST commands and eat to WDG command etc)
+    //TODO: Maybe a HandleEvtQueue instead that takes in the whole queue and eats the whole thing in order of non-blocking to blocking
+
+    //Switch for the GLOBAL_COMMAND
+    switch (cm.GetCommand()) {
+    case REQUEST_COMMAND: {
+        HandleRequestCommand(cm.GetTaskCommand());
+    }
+    case TASK_SPECIFIC_COMMAND: {
+        break;
+    }
+    default:
+        SOAR_PRINT("BatteryTask - Received Unsupported Command {%d}\n", cm.GetCommand());
+        break;
+    }
+
+    //No matter what we happens, we must reset allocated data
+    cm.Reset();
+}
+
+/**
+ * @brief Handles a Request Command
+ * @param taskCommand The command to handle
+ */
+void BatteryTask::HandleRequestCommand(uint16_t taskCommand)
+{
+    //Switch for task specific command within DATA_COMMAND
+    switch (taskCommand) {
+    case BATTERY_REQUEST_NEW_SAMPLE:
+        SampleBatteryVoltage();
+        GetPowerState();
+        break;
+    case BATTERY_REQUEST_TRANSMIT:
+    	TransmitProtocolBatteryData();
+        break;
+    case BATTERY_REQUEST_DEBUG:
+        SOAR_PRINT("|VOLTAGE_TASK| Battery Voltage (V): %d.%d, MCU Timestamp: %u\r\n", data->voltage_ / 1000, data->voltage_ % 1000,
+        timestampPT);
+        SOAR_PRINT("Battery Power State: %d, \r\n", GetPowerState());
+        break;
+    default:
+        SOAR_PRINT("UARTTask - Received Unsupported REQUEST_COMMAND {%d}\n", taskCommand);
+        break;
+    }
+}
+
+/**
+ * @brief This function reads and updates battery voltage reading
+ *          from the battery.
+ */
+void BatteryTask::SampleBatteryVoltage()
+{
+	static const int BATTERY_VOLTAGE_ADC_POLL_TIMEOUT = 50;
+	double adcVal[1] = {};
+	double batteryVoltageValue = 0;
+	double vi = 0;
+
+	/* Functions -----------------------------------------------------------------*/
+	HAL_ADC_Start(&hadc2);  // Enables ADC and starts conversion of regular channels
+	if(HAL_ADC_PollForConversion(&hadc2, BATTERY_VOLTAGE_ADC_POLL_TIMEOUT) == HAL_OK) { //Check if conversion is completed
+		adcVal[0] = HAL_ADC_GetValue(&hadc2); // Get ADC Value
+		HAL_ADC_Stop(&hadc2);
+		}
+
+	vi = ((3.3/4095) * (adcVal[0])); // Converts 12 bit ADC value into voltage
+	batteryVoltageValue = (vi * 4) * 1000; // Multiply by 1000 to keep decimal places
```

Comment:
do we know what this multiply by 4 does?

nit: should be a constexpr with a name describing what it does.